care/emr/api/viewsets/medication_request.py

```
@@ -41,5 +46,29 @@ def get_queryset(self):
+                raise PermissionDenied(
+                    "Requester does not have permission to update encounter"
+                )
+
+    def authorize_update(self, request_obj, model_instance):
+        super().authorize_update(request_obj, model_instance)
+        if model_instance.requester:
+            if not AuthorizationController.call(
+                "can_update_encounter_obj",
+                model_instance.requester,
+                model_instance.encounter,
+            ):
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)